### DISCUSSION TITLE
Unable to initialize OpenGL loader

### DISCUSSION THREAD

**Discussion by AdrianPeh** [ocornut/imgui#6814] [New Users Build/Link/Run issues ONLY!]
I have been following https://github.com/ocornut/imgui/wiki/Getting-Started#setting-up-dear-imgui--backends and the example codes to incorporate imgui into my school project. First time using imgui and not sure what is missing in my code. Tried looking around for past posts/discussions but non seem to be working. Looking for advice or feedback, will be looking on my own end too. Thanks for reading this post.

Issue: I am getting an assert from ImGui_ImplOpenGL3_NewFrame()
Potential cause: ImGui_ImplOpenGL3_Init("#version 450"); seems to not work

IDE: VS 2022
Renderer: OpenGL
Platform: Win32

The following snippets are what I have added/done so far

// Includes & headers
`
#include "imgui_impl_win32.h"
#include "imgui_impl_opengl3.h"
extern IMGUI_IMPL_API LRESULT ImGui_ImplWin32_WndProcHandler(HWND hWnd, UINT msg, WPARAM wParam, LPARAM lParam);
`

// WndProc handler

```
extern IMGUI_IMPL_API LRESULT ImGui_ImplWin32_WndProcHandler(HWND hWnd, UINT msg, WPARAM wParam, LPARAM lParam);

LRESULT CALLBACK WindowProc(HWND hwnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
{
    if (ImGui_ImplWin32_WndProcHandler(hwnd, uMsg, wParam, lParam))
        return true;

    switch (uMsg)
    {
    case WM_DESTROY:
        PostQuitMessage(0);
        le_gsm::nextState = le_gsm::GAMESTATE_QUIT;
        return 0;

    case WM_PAINT:
    {
        PAINTSTRUCT ps;
        HDC hdc = BeginPaint(hwnd, &ps);

        // All painting occurs here, between BeginPaint and EndPaint.
        FillRect(hdc, &ps.rcPaint, (HBRUSH)(COLOR_WINDOW + 1));
        EndPaint(hwnd, &ps);
        break;
    }
    default:
        break;
    }
    return DefWindowProc(hwnd, uMsg, wParam, lParam);
}
```

// Initialization

```
IMGUI_CHECKVERSION();
ImGui::CreateContext();
ImGuiIO& io = ImGui::GetIO();
io.ConfigFlags |= ImGuiConfigFlags_NavEnableKeyboard;     // Enable Keyboard Controls
io.ConfigFlags |= ImGuiConfigFlags_NavEnableGamepad;      // Enable Gamepad Controls
io.ConfigFlags |= ImGuiConfigFlags_DockingEnable;         // IF using Docking Branch
io.ConfigFlags |= ImGuiConfigFlags_ViewportsEnable;       // Enable Multi-Viewport / Platform Windows
// Setup Platform/Renderer backends
ImGui_ImplWin32_Init(windowHandle);
ImGui_ImplOpenGL3_Init("#version 450"); 
```

// Core loop
```
// Start dear imgui frame
ImGui_ImplOpenGL3_NewFrame();
ImGui_ImplWin32_NewFrame();
ImGui::NewFrame();

// Display window
ImGui::ShowDemoWindow();

// Rendering
ImGui::Render();
ImGui_ImplOpenGL3_RenderDrawData(ImGui::GetDrawData());

```
// Shutdown

```
ImGui_ImplOpenGL3_Shutdown();
ImGui_ImplWin32_Shutdown();
ImGui::DestroyContext();
```

**Comment by ocornut** [comment]
You are not specifying what the problem is. Are you getting an error, an assert, a crash?

Using GLSL version 450 requires a GL 4.5 contexts so that's entirely a question of how you initialize your GL context and has nothing to do with the OpenGL loader per-se, afaik.

**Reply by nicolasnoble** [reply]
Yeah, in general it's better to leave this string to nullptr so the ImGui backend can pick whichever is best.

**Reply by AdrianPeh** [reply]
Thanks for the feedback on details I should include in the discussion. I have made the necessary edits.

 Yup its an assert.
 I tried looking up context creation from https://open.gl/context but I don't seem to be able to get it running. Correct me if I'm wrong but since my platform/backend is already win32, there is no need for glfw right? Currently trying to follow https://learn.microsoft.com/en-us/windows/win32/opengl/rendering-context-functions to see if it helps.

**Reply by AdrianPeh** [reply]
Follow Up^^

Playing around with information from https://learn.microsoft.com/en-us/windows/win32/opengl/rendering-context-functions helped. The assert is not appearing, so I'm assuming the context is initialized correctly, followed by initialing the loader. The demo is not appearing but that's for a different post/discussion should I need help on it. 

Thanks for the help and advice! Pointed me in the right direction. Is there specific follow-up needed to be done for a proper closing of a discussion?
